stringdtype/stringdtype/src/dtype.h

```diff
@@ -26,6 +26,9 @@ new_stringdtype_instance(void);
 int
 init_string_dtype(void);
 
+int
+compare_strings(char **a, char **b, PyArrayObject *arr);
+
```

Comment:
I don't think this needs to be in the header, since its only usage is inside `dtype.c`.

Reply:
You could add a forward declaration inside that file, or just move its declaration above its usage.